kord-extensions/src/main/kotlin/com/kotlindiscord/kord/extensions/commands/application/ApplicationCommandRegistry.kt

```diff
@@ -605,124 +355,18 @@ public open class ApplicationCommandRegistry : KoinComponent {
 
             return null
         }
-
-        userCommands[response.id] = command
-
         return command
     }
 
-    // endregion
-
-    // region: Unregistration functions
-
-    /** Unregister a message command. **/
-    public open suspend fun unregisterGeneric(
-        command: ApplicationCommand<*>,
-        delete: Boolean = true
-    ): ApplicationCommand<*>? =
-        when (command) {
-            is MessageCommand<*> -> unregister(command, delete)
-            is SlashCommand<*, *> -> unregister(command, delete)
-            is UserCommand<*> -> unregister(command, delete)
-
-            else -> error("Unsupported application command type: ${command.type.name}")
-        }
-
-    /** Unregister a message command. **/
-    public open suspend fun unregister(command: MessageCommand<*>, delete: Boolean = true): MessageCommand<*>? {
-        val filtered = messageCommands.filter { it.value == command }
-        val id = filtered.keys.firstOrNull() ?: return null
-
-        if (delete) {
-            deleteGeneric(command, id)
-        }
-
-        return messageCommands.remove(id)
-    }
-
-    /** Unregister a slash command. **/
-    public open suspend fun unregister(command: SlashCommand<*, *>, delete: Boolean = true): SlashCommand<*, *>? {
-        val filtered = slashCommands.filter { it.value == command }
-        val id = filtered.keys.firstOrNull() ?: return null
-
-        if (delete) {
-            deleteGeneric(command, id)
-        }
-
-        return slashCommands.remove(id)
-    }
-
-    /** Unregister a user command. **/
-    public open suspend fun unregister(command: UserCommand<*>, delete: Boolean = true): UserCommand<*>? {
-        val filtered = userCommands.filter { it.value == command }
-        val id = filtered.keys.firstOrNull() ?: return null
-
-        if (delete) {
-            deleteGeneric(command, id)
-        }
-
-        return userCommands.remove(id)
-    }
-
-    /** @suppress Internal function used to delete the given command from Discord. Used by [unregister]. **/
-    public open suspend fun deleteGeneric(
-        command: ApplicationCommand<*>,
-        discordCommandId: Snowflake,
+    protected fun injectRawPermissions(
```

Comment:
This should also be open. In general, don't hide methods that will ever be useful in subtypes in any way - make them public, document them, and allow subtypes to use them.